nifi-nar-bundles/nifi-elasticsearch-bundle/nifi-elasticsearch-processors/src/main/java/org/apache/nifi/processors/elasticsearch/AbstractElasticsearchProcessor.java

```diff
@@ -0,0 +1,227 @@
+
+import org.apache.nifi.annotation.lifecycle.OnStopped;
+import org.apache.nifi.components.PropertyDescriptor;
+import org.apache.nifi.components.Validator;
+import org.apache.nifi.logging.ProcessorLog;
+import org.apache.nifi.processor.AbstractProcessor;
+import org.apache.nifi.processor.ProcessContext;
+import org.apache.nifi.processor.exception.ProcessException;
+import org.apache.nifi.processor.util.StandardValidators;
+import org.apache.nifi.ssl.SSLContextService;
+import org.apache.nifi.util.StringUtils;
+import org.elasticsearch.client.Client;
+import org.elasticsearch.client.transport.TransportClient;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.transport.InetSocketTransportAddress;
+
+import java.net.InetSocketAddress;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicReference;
+
+
+public abstract class AbstractElasticsearchProcessor extends AbstractProcessor {
+
+    protected static final PropertyDescriptor CLUSTER_NAME = new PropertyDescriptor.Builder()
+            .name("Cluster Name")
+            .description("Name of the ES cluster (for example, elasticsearch_brew). Defaults to 'elasticsearch'")
+            .required(false)
+            .addValidator(Validator.VALID)
+            .defaultValue("elasticsearch")
+            .build();
+    protected static final PropertyDescriptor HOSTS = new PropertyDescriptor.Builder()
+            .name("ElasticSearch Hosts")
+            .description("ElasticSearch Hosts, which should be comma separated and colon for hostname/port "
+                    + "host1:port,host2:port,....  For example testcluster:9200.")
+            .required(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor PROP_SSL_CONTEXT_SERVICE = new PropertyDescriptor.Builder()
+            .name("SSL Context Service")
+            .description("The SSL Context Service used to provide client certificate information for TLS/SSL "
+                    + "connections. This service only applies if the Shield plugin is available.")
+            .required(false)
+            .identifiesControllerService(SSLContextService.class)
+            .build();
+
+    public static final PropertyDescriptor PROP_SHIELD_LOCATION = new PropertyDescriptor.Builder()
+            .name("Shield plugin URL")
+            .description("Specifies the location of the JAR for the Elasticsearch Shield plugin."
+                    + " If the Elasticsearch cluster has been secured with the Shield plugin, then "
+                    + "an SSL Context Service must be defined, and the Shield plugin JAR must also "
+                    + "be available to this processor.")
+            .required(false)
+            .addValidator(StandardValidators.URL_VALIDATOR)
```

Comment:
I believe we should force this to be a file path instead of a URL so there isn't a network call out to grab the jar every time the client needs to restart.